src/main/java/com/epam/esm/service/util/impl/CertificateValidatorImpl.java

```diff
@@ -0,0 +1,91 @@
+package com.epam.esm.service.util.impl;
+
+import com.epam.esm.dao.request.CertificateSearchCriteria;
+import com.epam.esm.dao.sort.SortBy;
+import com.epam.esm.model.GiftCertificate;
+import com.epam.esm.service.exception.ErrorCodeEnum;
+import com.epam.esm.service.exception.ServiceException;
+import com.epam.esm.service.util.CertificateValidator;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+
+@Component
+public class CertificateValidatorImpl implements CertificateValidator {
+
+    @Override
+    public void validate(GiftCertificate giftCertificate) throws ServiceException {
```

Comment:
Now deleted all custom validators except Order validator. Need order validator because we have condition, that sum of certificate prices in order should be equal to price of order. And to check this condition is correct or not we need custom validator.